test/pylib/report_plugin.py

```diff
@@ -1,7 +1,24 @@
+import importlib
+import logging
+
 import pytest
 
+logger = logging.getLogger(name=__name__)
+_allure_module = None
+
+
+def _get_allure_module():
+    global _allure_module
+    if _allure_module is None:
+        _allure_module = importlib.import_module('allure')
+    return _allure_module
+
```

Comment:
What's the reason for all these games? Can't we require users to install allure? It's in install-dependencies.sh after all.

Reply:
If a random library is not *really* needed by the test, it is good form not to require it.

Reply:
@avikivity  it is according to request from @nyh  https://github.com/scylladb/scylladb/pull/19942#discussion_r1698318329

Reply:
I can understand not wanting fancy output (is that what it does? I'm still not sure), but we don't need to uglify the code in order to avoid having to install some package.

Reply:
It adds a possibility to generate better report with test results. So it's on touching the existing output.

Reply:
on=no?

So this patch is a no-op at present?

I don't think we need to make the dependency optional. In that direction lies madness, we'll have a matrix of possibly-installled-possibly-not dependencies and random code to handle that.

Reply:
allure-pytest module is already in toolchain. So this patch will switch it on for everyone. I'll rewrite to have ordinary import. @avikivity can I say that we are aiming to compatibility with toolchain? If someone is using their own host, it's their responsibility to install all dependencies?

Reply:
That's exactly correct.

Reply:
> on=no?
> 
> So this patch is a no-op at present?
> 
> I don't think we need to make the dependency optional. In that direction lies madness, we'll have a matrix of possibly-installled-possibly-not dependencies and random code to handle that.

l disagree. I think madness lies in forcing developers to install more and more obscure libraries that are only needed by CI processes and UI. And morever code that slow down tests and cause them to write additional output.

I need test/alternator/run to continue running, continue to run quickly, and not generate tons of unnecessary output.

Reply:
I guess I lost, and "allure" has now become a non-optional prerequisite to running a single test on developers' machines despite not being relevant to them at all. So we can resolve this discussion.

Reply:
@xtrey please make sure there is no runtime regression

Reply:
The current patch
  tools/toolchain/dbuild ./test.py topology_custom alternator  0.76s user 0.67s system 0% cpu 24:19.28 total
  tools/toolchain/dbuild ./test.py topology_custom alternator  0.74s user 0.60s system 0% cpu 22:00.27 total
  tools/toolchain/dbuild ./test.py topology_custom alternator  0.75s user 0.58s system 0% cpu 22:07.07 total

master branch
  tools/toolchain/dbuild ./test.py topology_custom alternator  1.08s user 0.90s system 0% cpu 23:58.85 total
  tools/toolchain/dbuild ./test.py topology_custom alternator  1.15s user 0.92s system 0% cpu 26:20.91 total
  tools/toolchain/dbuild ./test.py topology_custom alternator  1.18s user 0.95s system 0% cpu 26:35.07 total

Didn't execute all tests since it will be time-consuming. But in general, I do not see any degradation in performance.